test/timeline.js

```diff
@@ -0,0 +1,90 @@
+// const chai = require('chai')
+// const expect = chai.expect
+const nockHelper = require('./helpers/nock_helper')
+
+const Hubspot = require('..')
+
+describe('timeline', function() {
+  let hubspot
+
+  before(() => {
```

Comment:
I think this is a step in the right direction. Instead of refreshing the token we could write these tests on the assumption that you already have a un-expired token. I was thinking something along the lines of:

```js
before(() => {
  mockOauthEndpoint(
    /\/integrations\/v1\/[0-9]+\/timeline\/event-types'/,
    {
      id: 261,
      name: 'Test Event Type',
      headerTemplate:
        '# Title for event {{id}}\nThis is an event for {{objectType}}',
      detailTemplate:
        'This event happened on {{#formatDate timestamp}}{{/formatDate}}',
      applicationId: 123,
      objectType: 'CONTACT',
    },
  )

  hubspot = new Hubspot({ accessToken: process.env.ACCESS_TOKEN || "some-fake-token" })
}
after(nockHelper.resetNock)
```

You could then hit the real api in the test with an access token with something like:

```
NOCK_OFF=true ACCESS_TOKEN=my-super-long-and-real-access-token npm run mocha -- --grep "broadcasts"
```

Reply:
For this to work you would need something like the following in nockHelper:

```js
// test/nock_helper.js
const nock = require('nock')

class NockHelper {
  mockRateLimit() {
    nock('http://api.hubapi.com', { encodedQueryParams: true })
      .get('/integrations/v1/limit/daily')
      .query({ hapikey: 'demo' })
      .reply(200, [
        {
          name: 'api-calls-daily',
          usageLimit: 160000,
          currentUsage: 19742,
          collectedAt: Date.now(),
          fetchStatus: 'CACHED',
          resetsAt: Date.now() + 24 * 60 * 60 * 1000,
        },
      ])
  }

  mockEndpoint(path, data) {
    return () => {
      nock.disableNetConnect()
      this.mockRateLimit()
      nock('http://api.hubapi.com', { encodedQueryParams: true })
        .get(path)
        .query({ hapikey: 'demo' })
        .reply(200, data)
    }
  }

  mockOauthEndpoint(path, data) {
    return () => {
      nock.disableNetConnect()
      nock('http://api.hubapi.com')
        .get(path)
        .reply(200, data)
    }
  }

  resetNock() {
    nock.cleanAll()
    nock.enableNetConnect()
  }
}

module.exports = new NockHelper()